components/hab/src/cli.rs

```diff
@@ -115,232 +216,216 @@ pub enum UserKey {
-            (@setting SubcommandRequiredElseHelp)
-            (subcommand: PlanInit::clap().aliases(&["i", "in", "ini"]))
-            (subcommand: PlanRender::clap().aliases(&["r", "re", "ren", "rend", "rende"]))
-        )
-        (@subcommand ring =>
-            (about: "Commands relating to Habitat rings")
-            (aliases: &["r", "ri", "rin"])
-            (@setting ArgRequiredElseHelp)
-            (@setting SubcommandRequiredElseHelp)
-            (@subcommand key =>
-                (about: "Commands relating to Habitat ring keys")
-                (aliases: &["k", "ke"])
-                (@setting ArgRequiredElseHelp)
-                (@setting SubcommandRequiredElseHelp)
-                (subcommand: KeyExport::clap().aliases(&["e", "ex", "exp", "expo", "expor"]))
-                (subcommand: KeyImport::clap().aliases(&["i", "im", "imp", "impo", "impor"]))
-                (subcommand: RingKeyGenerate::clap().aliases(&["g", "ge", "gen", "gene", "gener", "genera", "generat"]))
-            )
-        )
-        (subcommand: HabSup::clap())
-        (@subcommand svc =>
-            (about: "Commands relating to Habitat services")
-            (aliases: &["sv", "ser", "serv", "service"])
-            (@setting ArgRequiredElseHelp)
-            (@setting SubcommandRequiredElseHelp)
-            (subcommand: SvcBulkLoad::clap())
-            (@subcommand key =>
-                (about: "Commands relating to Habitat service keys")
-                (aliases: &["k", "ke"])
-                (@setting ArgRequiredElseHelp)
-                (@setting SubcommandRequiredElseHelp)
-                (subcommand: KeyGenerate::clap().aliases(&["g", "ge", "gen", "gene", "gener", "genera", "generat"]))
-            )
-            (subcommand: SvcLoad::clap())
-            (subcommand: SvcUpdate::clap())
-            (subcommand: SvcStart::clap().aliases(&["star"]))
-            (subcommand: SvcStatus::clap().aliases(&["stat", "statu"]))
-            (subcommand: SvcStop::clap().aliases(&["sto"]))
-            (subcommand: SvcUnload::clap().aliases(&["u", "un", "unl", "unlo", "unloa"]))
-        )
-        (subcommand: Studio::clap().aliases(&["stu", "stud", "studi"]))
-        (@subcommand supportbundle =>
-            (about: "Create a tarball of Habitat Supervisor data to send to support")
-            (aliases: &["supp", "suppo", "suppor", "support-bundle"])
-        )
-        (@subcommand user =>
-            (about: "Commands relating to Habitat users")
-            (aliases: &["u", "us", "use"])
-            (@setting ArgRequiredElseHelp)
-            (@setting SubcommandRequiredElseHelp)
-            (subcommand: UserKey::clap())
-        )
-        (subcommand: alias_apply)
-        (subcommand: alias_install)
-        (subcommand: alias_run)
-        (subcommand: alias_setup)
-        (subcommand: alias_start)
-        (subcommand: alias_stop)
-        (subcommand: alias_term)
-        (after_help: AFTER_HELP)
-    )
+    Options::clap()
```

Comment:
Is this not the same as `Hab::clap()` above?

Reply:
Yup, here is the updated code.
```
pub fn get(feature_flags: FeatureFlag) -> App<'static, 'static> {
    if feature_flags.contains(FeatureFlag::STRUCTOPT_CLI) {
        return Hab::clap();
    }

    Options::clap()
}
```

Reply:
Originally, when we were first rolling out `structopt` to represent parts of the CLI, we hid it behind a feature flag so users could opt in and give feedback, but if anything went wrong, there was an "escape hatch".

Now, however, the CLI is only going to be implemented using `structopt`. In that case, it seems that `Hab::clap()` and `Options::clap()` are going to be the same. Thus, `Options` is basically a duplicate of `Hab`, which means we don't need `Options` at all. It furthermore implies that we can remove the `STRUCTOPT_CLI` feature flag altogether.

Are there meaningful differences between `Hab` and `Options`, or can the latter be removed?

Reply:
I see some **aliases** are being explicitly set on certain subcommands, which are not present in **Hab::clap()** part, and also see some settings are being set **settings = &[AppSettings::ArgRequiredElseHelp, AppSettings::SubcommandRequiredElseHelp]**